lib/evm_database_ops.rb

```diff
@@ -114,7 +114,13 @@ def self.restore(db_opts, connect_opts = {})
         connect_opts[:uri] = File.dirname(connect_opts[:uri])
       else
         connect_opts[:remote_file_name] ||= File.basename(backup_file_name(action))
-        backup_folder = action == :dump ? "db_dump" : "db_backup"
+        backup_folder = if connect_opts[:skip_directory]
+                          []
+                        elsif action == :dump
+                          "db_dump"
+                        else
+                          "db_backup"
+                        end
```

Comment:
I think this is my only suggestion.  Just don't like how this looks currently (question if my suggestion is better...), but I am a bit squeamish about passing a `[]` to `File.join` (seems odd):

```ruby
folder   = nil
folder ||= action == :dump ? "db_dump" : "db_backup" unless connect_opts[:skip_directory]
uri = File.join(*[connect_opts[:uri], folder, connect_opts[:remote_file_name].compact)
```

I would maybe suggest moving this to a method, but since we are doing class methods here (not necessarily a fan of how this class is built) we would have to pass `action` and `connect_opts` as well, so I did it this way for less complexity.

Reply:
```ruby
File.join("a", %w(b c), nil, "d")
TypeError: no implicit conversion of nil into String
	from (irb):1:in `join'
	from (irb):1
	from /opt/rubies/ruby-2.3.7/bin/irb:11:in `<main>'
File.join("a", %w(b c), [], "d")
# => "a/b/c/d"  
```

Reply:
I put in a work around, hope this one looks better.

Reply:
@kbrock just an FYI, I was using the `*` (splat) operator on a compacted array (removes `nil` values), so that is why my sample worked, where your example does not.

But, I think what you have changed it to works for me, so lets go with that.  Thanks.  👍